### ISSUE TITLE
[EKS addon] [CoreDNS] [request]: Add tolerations to CoreDNS addon config

### ISSUE THREAD

**Issue by RuStyC0der** [aws/containers-roadmap#1930]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
It would be great to add toleration settings to CoreDNS addon configuration. I have requested config shema for coredns plugin using eksctl (eksctl utils describe-addon-configuration --name coredns --version v1.8.7-eksbuild.3)  it has config for resources/replica count/node selector/etc but not tolerations. Can we add tolerations too?

**Which service(s) is this request for?**
EKS (CoreDNS add-on)

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
In project I working for we have fully tainted cluster to prevent misscheduling. It created with eksctl and every time creation finished coreDNS addon pods can not be scheduled because of node taints.

**Are you currently working around this issue?**
Curently I just manually patch coredns deployment but it not seems to be production-like solution.

**Additional context**
Please let me know  if I can help you with this feature in any way.

**Comment by youwalther65** [comment]
I just checked coredns add-on version v1.9.3-eksbuild.3 and the schema has changed a lot compared to previous version v1.9.3-eksbuild.2 and now contains affinity and tolerations! See here:

```yaml
$ aws eks describe-addon-configuration --addon-name coredns --addon-version v1.9.3-eksbuild.3 --query 'configurationSchema' --output text | jq
{
  "$ref": "#/definitions/Coredns",
  "$schema": "http://json-schema.org/draft-06/schema#",
  "definitions": {
    "Coredns": {
      "additionalProperties": false,
      "properties": {
        "affinity": {
          "default": {
            "affinity": {
              "nodeAffinity": {
                "requiredDuringSchedulingIgnoredDuringExecution": {
                  "nodeSelectorTerms": [
                    {
                      "matchExpressions": [
                        {
                          "key": "kubernetes.io/os",
                          "operator": "In",
                          "values": [
                            "linux"
                          ]
                        },
                        {
                          "key": "kubernetes.io/arch",
                          "operator": "In",
                          "values": [
                            "amd64",
                            "arm64"
                          ]
                        }
                      ]
                    }
                  ]
                }
              },
              "podAntiAffinity": {
                "preferredDuringSchedulingIgnoredDuringExecution": [
                  {
                    "podAffinityTerm": {
                      "labelSelector": {
                        "matchExpressions": [
                          {
                            "key": "k8s-app",
                            "operator": "In",
                            "values": [
                              "kube-dns"
                            ]
                          }
                        ]
                      },
                      "topologyKey": "kubernetes.io/hostname"
                    },
                    "weight": 100
                  }
                ]
              }
            }
          },
          "description": "Affinity of the coredns pods",
          "type": [
            "object",
            "null"
          ]
        },
        "computeType": {
          "type": "string"
        },
        "corefile": {
          "description": "Entire corefile contents to use with installation",
          "type": "string"
        },
        "nodeSelector": {
          "additionalProperties": {
            "type": "string"
          },
          "type": "object"
        },
        "replicaCount": {
          "type": "integer"
        },
        "resources": {
          "$ref": "#/definitions/Resources"
        },
        "tolerations": {
          "default": [
            {
              "key": "CriticalAddonsOnly",
              "operator": "Exists"
            },
            {
              "key": "node-role.kubernetes.io/master",
              "operator": "NoSchedule"
            }
          ],
          "description": "Tolerations of the coredns pod",
          "items": {
            "type": "object"
          },
          "type": "array"
        }
      },
      "title": "Coredns",
      "type": "object"
    },
    "Limits": {
      "additionalProperties": false,
      "properties": {
        "cpu": {
          "type": "string"
        },
        "memory": {
          "type": "string"
        }
      },
      "title": "Limits",
      "type": "object"
    },
    "Resources": {
      "additionalProperties": false,
      "properties": {
        "limits": {
          "$ref": "#/definitions/Limits"
        },
        "requests": {
          "$ref": "#/definitions/Limits"
        }
      },
      "title": "Resources",
      "type": "object"
    }
  }
}
```
It is first suppotrted with EKS v1.24
```yaml
$ aws eks describe-addon-versions --addon-name coredns --kubernetes-version 1.24 --query 'addons[*].addonVersions[*].addonVersion'
[
    [
        "v1.9.3-eksbuild.3",
        "v1.9.3-eksbuild.2",
        "v1.8.7-eksbuild.4",
        "v1.8.7-eksbuild.3",
        "v1.8.4-eksbuild.2"
    ]
]
```

Can someone please check and confirm.

**Comment by siddharthdeshmukh** [comment]
Hi,

Is this going to be released for EKS 1.23 too?

**Comment by sjastis** [comment]
As stated [here](https://docs.aws.amazon.com/eks/latest/userguide/managing-coredns.html), the support is available with 1.24+

**Comment by RuStyC0der** [comment]
As @youwalther65  mentioned version v1.9.3+ of coreDNS addon now support tolerations so we can close the issue. @youwalther65 Thank you for notifying about this new update.
